core/templates/pages/classroom-page/classroom-page-root.component.html

```diff
@@ -7,3 +7,5 @@
 
   </oppia-base-content>
 </oppia-angular-root>
+
+<oppia-error-page-root *ngIf="errorPageIsShown"></oppia-error-page-root>
```

Comment:
At any point, do we expect only one of these to show i.e. error page or the classroom page? If so, should we use just one variable and negate that for the "else" condition?

Reply:
Yes, we can do that but in other instances of the codebase, we have followed two variable patterns eg, blog-home-page, etc. I think this will make it consistent but I am also fine with the suggestion. What would you prefer? Thanks

Reply:
Yeah, I got a use case -- 
If `pageIsShown` is true -- classroom page is shown and when `pageIsShown` is false the 404 page is shown.
Initially `pageIsShown` is false. And by making a backend request we check whether the classroom config exists or not. If it exists then `pageIsShown` is assigned to true. Since the backend request is an Async call,  the time we get the data, it will temporarily show a 404 page, and then the classroom page is shown.

Sample video: 

https://github.com/oppia/oppia/assets/57531197/b4e1a1d3-dc0f-4169-91cf-1e92bcaf3e8c

Reply:
Makes sense, let's keep it in that case. Thanks!